### PR TITLE
SWATCH-1750: Configure docker in docker in Jenkins

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: [SWATCH-1750](https://issues.redhat.com/browse/SWATCH-1750)

### CODE REVIEW

**Comment by lindseyburnett** [comment]
TODO:

- [ ] see if we can find a RedHat source for kubedock

**Review by kahowell — CHANGES_REQUESTED** [review]

File: swatch-core-test/src/main/java/org/candlepin/containers/SwatchPostgreSQLContainer.java [hunk 1]
```diff
@@ -18,26 +18,40 @@
  * granted to use or replicate Red Hat trademarks that are incorporated
  * in this software or its documentation.
  */
-package org.candlepin.subscriptions.test;
+package org.candlepin.containers;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply swatch-core-test/src/main/java/org/candlepin/containers/SwatchPostgreSQLContainer.java]: done

File: Jenkinsfile [hunk 2]
```diff
@@ -5,20 +5,49 @@ pipeline {
...
-            }
-
             defaultContainer 'openjdk17'
+            yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+    - name: kubedock
+      image: docker.io/joyrex2001/kubedock:0.13.0
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply Jenkinsfile]: I've just created the MR to include this image: https://gitlab.cee.redhat.com/jcarvaja/app-interface/-/merge_requests/2

I also had to create the GH repository: https://github.com/RedHatInsights/kubedock

I verified this image with this pull request and everything looks good. 
Once the MR is merged and the image pushed, I will update this pull request.
For now, I'm changing this pull request to draft.

File: swatch-core-test/src/main/java/org/candlepin/containers/SwatchPostgreSQLContainer.java [hunk 3]
```diff
@@ -18,26 +18,40 @@
...
 public class SwatchPostgreSQLContainer extends PostgreSQLContainer<SwatchPostgreSQLContainer> {
-
-  /**
-   * This image is a replacement for the official one "docker.io/library/postgres:13" but in Quay.io
-   * to avoid Docker Hub exceed limit issues.
-   */
-  private static final String IMAGE = "quay.io/jcarvaja/postgres:13";
+  private static final String SWATCH_PGDATA_PROPERTY = "SWATCH_PGDATA";
+  private static final String POSTGRESQL_IMAGE_PROPERTY = "POSTGRESQL_IMAGE";
+  private static final String POSTGRESQL_IMAGE_DEFAULT = "docker.io/library/postgres:13";
```
[the remark this answers is not in the document]
Comment (kahowell) [reply swatch-core-test/src/main/java/org/candlepin/containers/SwatchPostgreSQLContainer.java]: Thanks!

**Review by kahowell — APPROVED** [review]
Thanks for addressing my feedback!

I'm okay to merge with your image as a stop-gap if you want to update it in a follow-up PR to one built by the new repo.

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (kahowell) [reply Jenkinsfile]: quay image was built: quay.io/cloudservices/kubedock:123efe0. (Automation doesn't provide a `latest` tag).

**Comment by Sgitario** [comment]
> @Sgitario Jenkins console logs shows 501 on docker api. Is it an issue ?
> 
> ```
> 
> 2023-09-24 00:14:38,475 [thread=Thread-24] [ERROR] [org.apache.zookeeper.server.NIOServerCnxnFactory] - Thread Thread[Thread-24,5,testcontainers] died
> com.github.dockerjava.api.exception.DockerException: Status 501: 
> 	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:249)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:124)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.PruneCmdExec.execute(PruneCmdExec.java:33)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.PruneCmdExec.execute(PruneCmdExec.java:13)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
> 	at org.testcontainers.shaded.com.github.dockerjava.core.command.PruneCmdImpl.exec(PruneCmdImpl.java:100)
> 	at org.testcontainers.utility.JVMHookResourceReaper.prune(JVMHookResourceReaper.java:58)
> 	at org.testcontainers.utility.JVMHookResourceReaper.lambda$performCleanup$2(JVMHookResourceReaper.java:27)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
> 	at org.testcontainers.utility.JVMHookResourceReaper.performCleanup(JVMHookResourceReaper.java:27)
> 	at java.base/java.lang.Thread.run(Thread.java:833)
> ```

I see this is happening when in the shutdown hook by Spring:

``` 
2023-09-25 18:35:44,307 [thread=SpringApplicationShutdownHook] [INFO ] [org.candlepin.subscriptions.task.queue.kafka.KafkaApplicationListener] - Shutting down kafka consumers...
2023-09-25 18:35:44,405 [thread=Thread-136] [INFO ] [org.candlepin.subscriptions.task.queue.inmemory.ExecutorTaskProcessor] - Stopping in-memory task processor
2023-09-25 18:35:44,556 [thread=Thread-24] [ERROR] [org.apache.zookeeper.server.NIOServerCnxnFactory] - Thread Thread[Thread-24,5,testcontainers] died
com.github.dockerjava.api.exception.DockerException: Status 501: 
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:249)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:124)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.PruneCmdExec.execute(PruneCmdExec.java:33)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.PruneCmdExec.execute(PruneCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.PruneCmdImpl.exec(PruneCmdImpl.java:100)
	at org.testcontainers.utility.JVMHookResourceReaper.prune(JVMHookResourceReaper.java:58)
	at org.testcontainers.utility.JVMHookResourceReaper.lambda$performCleanup$2(JVMHookResourceReaper.java:27)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testcontainers.utility.JVMHookResourceReaper.performCleanup(JVMHookResourceReaper.java:27)
	at java.base/java.lang.Thread.run(Thread.java:833)
2023-09-25 18:35:44,629 [thread=Thread-155] [INFO ] [org.candlepin.subscriptions.task.queue.inmemory.ExecutorTaskProcessor] - Stopping in-memory task processor
``` 

Since it's a 501, I assume it's caused because zookeeper is already terminated, so I don't think it's relevant.

**Pull request merged**